WordPress/src/main/java/org/wordpress/android/ui/jetpack/JetpackCapabilitiesUseCase.kt

```diff
@@ -31,58 +30,62 @@ class JetpackCapabilitiesUseCase @Inject constructor(
     @Suppress("unused") private val siteStore: SiteStore,
     private val dispatcher: Dispatcher,
     private val appPrefsWrapper: AppPrefsWrapper,
-    private val currentDateProvider: CurrentTimeProvider,
-    @param:Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+    private val currentDateProvider: CurrentTimeProvider
 ) {
-    private var continuation: Continuation<OnJetpackCapabilitiesFetched>? = null
+    private var continuation: HashMap<Long, Continuation<OnJetpackCapabilitiesFetched>?> = hashMapOf()
 
-    suspend fun getJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts {
-        val capabilities = withContext(bgDispatcher) {
-            getOrFetchJetpackCapabilities(remoteSiteId)
+    suspend fun getJetpackPurchasedProducts(remoteSiteId: Long) = flow {
+        emit(getCachedJetpackPurchasedProducts(remoteSiteId))
+        if (!hasValidCache(remoteSiteId)) {
+            emit(fetchJetpackPurchasedProducts(remoteSiteId))
         }
-        return JetpackPurchasedProducts(
-                scan = capabilities.find { SCAN_CAPABILITIES.contains(it) } != null,
-                backup = capabilities.find { BACKUP_CAPABILITIES.contains(it) } != null
-        )
     }
 
-    suspend fun getOrFetchJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return if (hasValidCache(remoteSiteId)) {
-            getCachedJetpackCapabilities(remoteSiteId)
-        } else {
-            fetchJetpackCapabilities(remoteSiteId)
-        }
-    }
+    fun getCachedJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts =
+            mapToJetpackPurchasedProducts(getCachedJetpackCapabilities(remoteSiteId))
 
-    private suspend fun hasValidCache(remoteSiteId: Long): Boolean {
-        return withContext(bgDispatcher) {
-            val lastUpdated = appPrefsWrapper.getSiteJetpackCapabilitiesLastUpdated(remoteSiteId)
-            lastUpdated > currentDateProvider.currentDate().time - MAX_CACHE_VALIDITY
-        }
+    suspend fun fetchJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts =
+            mapToJetpackPurchasedProducts(fetchJetpackCapabilities(remoteSiteId))
+
+    private fun mapToJetpackPurchasedProducts(capabilities: List<JetpackCapability>) =
+            JetpackPurchasedProducts(
+                    scan = capabilities.find { SCAN_CAPABILITIES.contains(it) } != null,
+                    backup = capabilities.find { BACKUP_CAPABILITIES.contains(it) } != null
+            )
+
+    fun hasValidCache(remoteSiteId: Long): Boolean {
```

Comment:
This method was not changed at all - I'm not sure why github marked it as changed.

Reply:
Happens to me sometimes too .. I think it's AS .